File: TheMovie/app/src/main/java/com/nyinyi/nw/themovie/adapter/MovieListAdapter.java
package com.nyinyi.nw.themovie.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.nyinyi.nw.themovie.R;
import com.nyinyi.nw.themovie.activity.MovieDetailActivity;
import com.nyinyi.nw.themovie.util.MovieConstants;
import com.nyinyi.nw.themovie.vos.UpcomingVO;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by User on 9/14/2017.
 */

public class MovieListAdapter extends RecyclerView.Adapter<MovieListAdapter.MovieViewHolder> {

    private Context context;
    private List<UpcomingVO> upcomingVOList;

    public MovieListAdapter(Context context) {
        this.context = context;
        upcomingVOList = new ArrayList<>();
    }

    @Override
    public MovieViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.upcomingmovie, parent, false);
        return new MovieViewHolder(view);

    }

    @Override
    public void onBindViewHolder(final MovieViewHolder holder, final int position) {
        holder.bind(upcomingVOList.get(position));
        holder.ivMovie.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(context, MovieDetailActivity.class);
                intent.putExtra("id",upcomingVOList.get(position).getId().toString());
                intent.putExtra("title",upcomingVOList.get(position).getTitle());
                intent.putExtra("vote_average",upcomingVOList.get(position).getVoteAverage().toString());
                intent.putExtra("poster_path",upcomingVOList.get(position).getPosterPath());
                intent.putExtra("original_language",upcomingVOList.get(position).getOriginalLanguage());
                intent.putExtra("original_title",upcomingVOList.get(position).getOriginalTitle());
                intent.putExtra("backdrop_path",upcomingVOList.get(position).getBackdropPath());
                intent.putExtra("adult",upcomingVOList.get(position).getAdult());
                intent.putExtra("overview",upcomingVOList.get(position).getOverview());
                intent.putExtra("release_date",upcomingVOList.get(position).getReleaseDate());
                context.startActivity(intent);

            }
        });
    }

    @Override
    public int getItemCount() {
        return upcomingVOList.size();
    }

    public void setUpcomingVOList(List<UpcomingVO> upcomingVOList) {
        this.upcomingVOList = upcomingVOList;
        notifyDataSetChanged();
    }

    public class MovieViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.iv_movie)
        ImageView ivMovie;

        private UpcomingVO mData;

        public MovieViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);

        }

        public void bind(UpcomingVO data) {
            mData = data;

            Glide.with(context)
                    .load(MovieConstants.IMAGE_URL + data.getPosterPath())
                    .placeholder(R.drawable.images)
                    .error(R.drawable.images)
                    .into(ivMovie);


        }

    }
}

File: TheMovie/app/src/main/java/com/nyinyi/nw/themovie/adapter/MovieListAdapter.kt
package com.nyinyi.nw.themovie.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.nyinyi.nw.themovie.R
import com.nyinyi.nw.themovie.activity.MovieDetailActivity
import com.nyinyi.nw.themovie.util.MovieConstants
import com.nyinyi.nw.themovie.vos.UpcomingVO
import java.util.*

/**
 * Created by User on 9/14/2017.
 */

class MovieListAdapter(private val context: Context) : RecyclerView.Adapter<MovieListAdapter.MovieViewHolder>() {
    private var upcomingVOList: List<UpcomingVO>? = null

    init {
        upcomingVOList = ArrayList()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {

        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.upcomingmovie, parent, false)
        return MovieViewHolder(view)

    }

    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
        holder.bind(upcomingVOList!![position])
        holder.ivMovie!!.setOnClickListener { v ->
            val intent = Intent(context, MovieDetailActivity::class.java)
            intent.putExtra("id", upcomingVOList!![position].id!!.toString())
            intent.putExtra("title", upcomingVOList!![position].title)
            intent.putExtra("vote_average", upcomingVOList!![position].voteAverage!!.toString())
            intent.putExtra("poster_path", upcomingVOList!![position].posterPath)
            intent.putExtra("original_language", upcomingVOList!![position].originalLanguage)
            intent.putExtra("original_title", upcomingVOList!![position].originalTitle)
            intent.putExtra("backdrop_path", upcomingVOList!![position].backdropPath)
            intent.putExtra("adult", upcomingVOList!![position].adult)
            intent.putExtra("overview", upcomingVOList!![position].overview)
            intent.putExtra("release_date", upcomingVOList!![position].releaseDate)
            context.startActivity(intent)

        }
    }

    override fun getItemCount(): Int {
        return upcomingVOList!!.size
    }

    fun setUpcomingVOList(upcomingVOList: List<UpcomingVO>) {
        this.upcomingVOList = upcomingVOList
        notifyDataSetChanged()
    }

    inner class MovieViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {

        internal var ivMovie: ImageView? = null

        init {
            ivMovie = itemView.findViewById(R.id.iv_movie)

        }

        fun bind(data: UpcomingVO) {

            Glide.with(context)
                    .load(MovieConstants.IMAGE_URL + data.posterPath!!)
                    .apply(RequestOptions().placeholder(R.drawable.images).error(R.drawable.images))
                    .into(ivMovie!!)


        }

    }
}
